engine/src/flutter/shell/platform/linux/fl_renderer.cc

```diff
@@ -388,6 +417,53 @@ guint32 fl_renderer_get_fbo(FlRenderer* self) {
   return 0;
 }
 
+struct _FlRendererCallBlockingTrampolineData {
+  FlRenderer* renderer;
+  gboolean* finished;
+  void (*callback)(gpointer data);
+  gpointer data;
+};
+
+static void fl_renderer_call_blocking_trampoline(gpointer user_data) {
+  auto* data = static_cast<_FlRendererCallBlockingTrampolineData*>(user_data);
+  data->callback(data->data);
+  FlRendererPrivate* priv = reinterpret_cast<FlRendererPrivate*>(
+      fl_renderer_get_instance_private(data->renderer));
+  g_autoptr(GMutexLocker) locker = g_mutex_locker_new(&priv->mutex);
+  *data->finished = TRUE;
+  g_cond_signal(&priv->cond);
+}
+
+// Executes the given function on the main thread and blocks until it completes.
+static void fl_renderer_call_blocking(FlRenderer* self,
+                                      void (*callback)(gpointer data),
+                                      gpointer callback_data) {
```

Comment:
Since it is only used once, I find the fact that this function is generic to be a bit of an unnecessary indirection. Could we instead just provide:

```suggestion
// Presents layers on the main thread with the provided data and waits for
// the presentation to complete before calling [fl_renderer_present_layers_trampoline]
static void fl_renderer_present_layers_blocking(FlRenderer* self,,
                                      _FlRendererPresentLayersData* callback_data) {
```

And then always pass `fl_renderer_present_layers_trampoline` as the callback in `_FlRendererCallBlockingTrampolineData`?